protostar/testing/test_suite.py

```diff
@@ -7,11 +7,16 @@
 
 class TestCase:
     def __init__(
-        self, test_path: Path, test_fn_name: str, setup_fn_name: Optional[str] = None
+        self,
+        test_path: Path,
+        test_fn_name: str,
+        setup_fn_name: Optional[str] = None,
+        available_gas: Optional[int] = None,
```

Comment:
Can gas be added to `TestCaseWithOffsets` and rename it to `Cairo1TestCase` perhaps? it'd be cairo-1 specific then.